pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentOpenLongPairRangeSet.java

```diff
@@ -220,6 +222,35 @@ public void forEach(RangeProcessor<T> action, LongPairConsumer<? extends T> cons
         });
     }
 
+    @Override
+    public <O> void forEachWithRangeBoundMapper(LongPairConsumer<O> rawRangeBoundMapper,
```

Comment:
There is a lot of duplication of code with `forEach`, can it be optimized?

Reply:
thanks for review. the generic for range require for `<T extands Comparable<T>>` which is used in `Range<T>` but for iteration ops we may transform to non Comparable object. i try to reuse but we need the `MLDataFormats.MessageRange` implements `Comparable`

Reply:
Ichange the signature and optimize the code. please take a look~